src/server.ts

```diff
@@ -0,0 +1,87 @@
+import express from 'express';
+import { randomUUID } from 'crypto';
+import bodyParser from 'body-parser';
+import { createValidator } from 'express-joi-validation';
+import {
+  createUserBodySchema,
+  paramsIDSchema,
+  paramsSubstringLimitSchema,
+  updateUserBodySchema,
+} from './schemas';
+import type { User } from './types';
+
+const app = express();
+const validator = createValidator();
+
+const users: User[] = [];
+
+app.use(bodyParser.json());
+app.use(express.json());
+
+// get all users
+app.get('/users', (req, res) => {
+  res.json(users);
+});
+
+// get user by ID
+app.get('/user/:id', validator.params(paramsIDSchema), (req, res) => {
+  const filteredUsers = users.filter((user) => user.id === req.params.id);
+  if (filteredUsers.length) {
+    res.json(filteredUsers);
+  } else {
+    res.json(`no user with id ${req.params.id}`);
```

Comment:
If you send json, then it makes sense to send an object, not a string. And I believe you have to set a different status (it's 200 by default). Although probably you will learn later about statuses.

Reply:
fixed